build/integration-in-docker.sh

```diff
@@ -19,35 +19,57 @@ set -e
 ROOT="$(cd "$(dirname "${BASH_SOURCE}")/.." && pwd -P)"
 TMPDIR=$(mktemp -d)
 function delete() {
-    echo "Deleting ${TMPDIR}..."
-    if [[ $EUID -ne 0 ]]; then
-      sudo rm -rf "${TMPDIR}"
-    else
-      rm -rf "${TMPDIR}"
-    fi
+  echo "Deleting ${TMPDIR}..."
+  if [[ $EUID -ne 0 ]]; then
+    sudo rm -rf "${TMPDIR}"
+  else
+    rm -rf "${TMPDIR}"
+  fi
 }
 trap delete EXIT INT
 
-docker run --rm \
-  -w /go/src/github.com/google/cadvisor \
-  -v ${PWD}:/go/src/github.com/google/cadvisor \
-  golang:1.13 \
-  bash -c "env GOOS=linux GO_FLAGS='-race' ./build/build.sh amd64 && \
-    env GOOS=linux go test -c github.com/google/cadvisor/integration/tests/api &&  \
-    env GOOS=linux go test -c github.com/google/cadvisor/integration/tests/healthz"
-
-EXTRA_DOCKER_OPTS="-e DOCKER_IN_DOCKER_ENABLED=true"
-if [[ "${OSTYPE}" == "linux"* ]]; then
-  EXTRA_DOCKER_OPTS+=" -v ${TMPDIR}/docker-graph:/docker-graph"
-fi
-
-mkdir ${TMPDIR}/docker-graph
-docker run --rm \
-  -w /go/src/github.com/google/cadvisor \
-  -v ${ROOT}:/go/src/github.com/google/cadvisor \
-  ${EXTRA_DOCKER_OPTS} \
-  --privileged \
-  --entrypoint="" \
-  gcr.io/k8s-testimages/bootstrap \
-  bash -c "apt update && apt install sudo && \
-/usr/local/bin/runner.sh build/integration.sh"
+function reset() {
+  delete
+  TMPDIR=$(mktemp -d)
+}
+
+function run_tests() {
+  GO_FLAGS=${1:-}
+  PACKAGES=${2:-"sudo"}
+  BUILD_PACKAGES=${3:-}
+  CADVISOR_ARGS=${4:-}
+  BUILD_CMD="env GOOS=linux GO_FLAGS='$GO_FLAGS' ./build/build.sh amd64 && \
+    env GOOS=linux GOFLAGS='$GO_FLAGS' go test -c github.com/google/cadvisor/integration/tests/api && \
+    env GOOS=linux GOFLAGS='$GO_FLAGS' go test -c github.com/google/cadvisor/integration/tests/healthz"
+
+  if [ "$BUILD_PACKAGES" != "" ]; then
+    BUILD_CMD="apt-get update && apt-get install $BUILD_PACKAGES && \
+    $BUILD_CMD"
+  fi
+  docker run --rm \
+    -w /go/src/github.com/google/cadvisor \
+    -v ${PWD}:/go/src/github.com/google/cadvisor \
+    golang:1.13 \
+    bash -c "$BUILD_CMD"
+
+  EXTRA_DOCKER_OPTS="-e DOCKER_IN_DOCKER_ENABLED=true"
+  if [[ "${OSTYPE}" == "linux"* ]]; then
+    EXTRA_DOCKER_OPTS+=" -v ${TMPDIR}/docker-graph:/docker-graph"
+  fi
+
+  mkdir ${TMPDIR}/docker-graph
+  docker run --rm \
+    -w /go/src/github.com/google/cadvisor \
+    -v ${ROOT}:/go/src/github.com/google/cadvisor \
+    ${EXTRA_DOCKER_OPTS} \
+    --privileged \
+    --cap-add="sys_admin" \
+    --entrypoint="" \
+    gcr.io/k8s-testimages/bootstrap \
+    bash -c "apt update && apt install $PACKAGES && \
+CADVISOR_ARGS="$CADVISOR_ARGS" /usr/local/bin/runner.sh build/integration.sh"
+}
+
+run_tests "-race" "sudo"
```

Comment:
Is there a reason why we want to do perf tests separately?  Are they expected to fail in some scenarios?

Reply:
Just read the comment above.  I think this is a reasonable solution, although I think we should consider having a separate target.  That way users who just want to run the "distro agnostic" tests can do so without having errors mixed in.